allennlp/training/trainer.py

```diff
@@ -122,22 +122,29 @@ def __call__(
         epoch: int,
         batch_number: int,
         is_training: bool,
+        is_master: bool,
     ) -> None:
-        raise NotImplementedError
+        pass
+
+
+BatchCallback.register("null")(BatchCallback)
```

Comment:
Just FYI, if that check in `FromParams` that we added this for bothers us in the future, we should probably just remove it.  It's potentially brittle, and this is an easier solution.

Reply:
Maybe, but in this case it's easy to write a `null` implementation. Not all classes are like that. But I guess we can always `throw NotImplementedError()`.